controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -464,18 +458,44 @@ func (r *StorageConsumerReconciler) reconcileCephRadosNamespace(
 		}
 		rns.Namespace = r.namespace
 
-		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
-			if err := controllerutil.SetControllerReference(r.storageConsumer, rns, r.Scheme); err != nil {
-				return err
+		deleteRns := false
+		if !bp.DeletionTimestamp.IsZero() && bp.Status != nil {
```

Comment:
Please handle delete "after" normal reconcile code

Reply:
possible to expand, we either create/update rns or delete rns, if handling after normal, it'll be we reconcile update and then delete.

Reply:
done.